### DISCUSSION TITLE
Razor v1.9.77.0

### DISCUSSION THREAD

**Discussion by markdwags** [markdwags/Razor#211] [Announcements]
This version includes more updates to the scripting engine, including the ability to create cooldowns that will display as a gump. This version also introduces lists and timers to the scripting engine. Be sure to review the latest scripting docs for details.

* Added the `cooldown` command that generates a gump-based cooldown timer that can be customized (icon, color, name, order, etc)
* Added [list commands](https://www.razorce.com/guide/commands/#list-commands) and [list expressions](https://www.razorce.com/guide/expressions/#list-expressions)
* Added [timer commands](https://www.razorce.com/guide/commands/#timer-commands) and [timer expressions](https://www.razorce.com/guide/expressions.md/#timer-expressions)
* Added the `foreach` loop to interact with lists
* Option to disable 25ms delay on script line execution
* Updated script variables to be part of the same system that sets variables using the `as` operator
* Fixed missing syntax highlight for `maxweight`, `clearhands`, `cleardragdrop`, `sell`
* Running a script will show the name in the system message
* Fixed the `ignore` and `unignore` commands
* Fixed the `varexists` expression
* Added the `unsetvar` command
* Updated the `interrupt` command to let you pass a specific layer
* Option to disable highlighting on the script UI
* Better error handling on script syntax issues

<hr /><em>This discussion was created from the release <a href='https://github.com/markdwags/Razor/releases/tag/v1.9.77.0'>Razor v1.9.77.0</a>.</em>

**Comment by pacificpleb** [comment]
Thanks for keeping this alive!  Would it be possible to request for UOPS to have some of the same functionality as in game?
Specifically:
 - Free view (ability to click and drag map to see other areas that you aren't currently in)
 - Zoom

**Reply by markdwags** [reply]
Since that functionality already exists within ClassicUO (via the WorldMap), while UOPS has remained in Razor, it hasn't received any updates.

**Reply by pacificpleb** [reply]
It almost* exists in game, the big caveat being you can’t use the in game map on a second monitor easily like UOPS

**Reply by BrokeUOplayer** [reply]
Razor versions 1.0.14 and 1.9.77.0 are getting the following errors and UO launcher no longer works, UOL cannot find server. 



Using OSI with classic client just trying to start RAZOR (without UO Launcher).

This just occurred when UO performed their latest patch on 7/28

UO launcher no longer works period!

Windows 10 running X86

RAZOR version 1.0.14

There was an exception while attempting to load 'C:\Games\Electronic Arts\Ultima Online Classic\cliloc.ENU':
System.ArgumentOutOfRangeException: Non-negative number required.
Parameter name: count
at System.IO.BinaryReader.Read(Byte[] buffer, Int32 index, Int32 count)
at Ultima.StringList..ctor(String language)
at Assistant.Language.LoadCliLoc()

**Comment by mrSTyshchenko** [comment]
Thanks a lot for the update. Keep up the good work!

**Reply by BrokeUOplayer** [reply]
Cannot get this version to connect to OSI server, when trying to set up launcher, I cannot add the razor plugin, it just closes out launcher when trying to add it. How do I set this new version up to run on classic client with OSI server?
